go/pkg/chunker/chunker.go

```diff
@@ -21,6 +21,42 @@ var IOBufferSize = 10 * 1024 * 1024
 // ErrEOF is returned when Next is called when HasNext is false.
 var ErrEOF = errors.New("ErrEOF")
 
+type UploadEntry struct {
```

Comment:
Can you please separate this into a different package? My reasoning is: the chunker is something very particular that only a few things depend upon (only the actual upload code), but many more things need to depend on UploadEntry (rexec, for instance). So it's separation of concerns.